sonar-batch-protocol/src/main/java/org/sonar/batch/protocol/viewer/ViewerApplication.java

```diff
@@ -0,0 +1,341 @@
+      DefaultMutableTreeNode childNode = createNode(child);
+      parentNode.add(childNode);
+      loadChildren(child, childNode);
+    }
+
+  }
+
+  private void updateTitle() {
+    frame.setTitle(metadata.getProjectKey() + (metadata.hasBranch() ? (" (" + metadata.getBranch() + ")") : "") + " " + sdf.format(new Date(metadata.getAnalysisDate())));
+  }
+
+  private void updateDetails(Component component) {
+    componentEditor.setText(component.toString());
+    updateHighlighting(component);
+    updateSource(component);
+    updateCoverage(component);
+  }
+
+  private void updateCoverage(Component component) {
+    coverageEditor.setText("");
+    if (reader.hasCoverage(component.getRef())) {
+      try (CloseableIterator<BatchReport.Coverage> it = reader.readComponentCoverage(component.getRef())) {
+        while (it.hasNext()) {
+          BatchReport.Coverage coverage = it.next();
+          coverageEditor.getDocument().insertString(coverageEditor.getDocument().getEndPosition().getOffset(), coverage.toString() + "\n", null);
+        }
+      } catch (Exception e) {
+        throw new IllegalStateException("Can't read code coverage for " + getNodeName(component), e);
+      }
+    }
+  }
+
+  private void updateSource(Component component) {
+    File sourceFile = reader.getFileStructure().fileFor(Domain.SOURCE, component.getRef());
+    if (sourceFile.exists()) {
+      try (Scanner s = new Scanner(sourceFile, StandardCharsets.UTF_8.name()).useDelimiter("\\Z")) {
+        sourceEditor.setText(s.next());
+      } catch (IOException ex) {
+        StringWriter errors = new StringWriter();
+        ex.printStackTrace(new PrintWriter(errors));
+        sourceEditor.setText(errors.toString());
+      }
+    } else {
+      sourceEditor.setText("");
+    }
+  }
+
+  private void updateHighlighting(Component component) {
+    highlightingEditor.setText("");
+    if (reader.hasSyntaxHighlighting(component.getRef())) {
+      try (CloseableIterator<BatchReport.SyntaxHighlighting> it = reader.readComponentSyntaxHighlighting(component.getRef())) {
+        while (it.hasNext()) {
+          BatchReport.SyntaxHighlighting rule = it.next();
+          highlightingEditor.getDocument().insertString(highlightingEditor.getDocument().getEndPosition().getOffset(), rule.toString() + "\n", null);
+        }
+      } catch (Exception e) {
+        throw new IllegalStateException("Can't read syntax highlighting for " + getNodeName(component), e);
+      }
+    }
+  }
+
+  /**
+   * Create the application.
+   */
+  public ViewerApplication() {
+    initialize();
+  }
+
+  /**
+   * Initialize the contents of the frame.
+   */
+  private void initialize() {
+    try {
+      for (LookAndFeelInfo info : UIManager.getInstalledLookAndFeels()) {
+        if ("Nimbus".equals(info.getName())) {
+          UIManager.setLookAndFeel(info.getClassName());
+          break;
+        }
+      }
+    } catch (Exception e) {
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png) Either log or rethrow this exception. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1166)